source/OutfitterPanel.cpp

```diff
@@ -309,49 +309,164 @@ int OutfitterPanel::DrawDetails(const Point &center)
 
 
 
-bool OutfitterPanel::CanBuy(bool checkAlreadyOwned) const
+ShopPanel::BuyResult OutfitterPanel::CanBuy(bool onlyOwned) const
 {
 	if(!planet || !selectedOutfit)
 		return false;
 
-	bool isAlreadyOwned = checkAlreadyOwned && IsAlreadyOwned();
-	if(!(outfitter.Has(selectedOutfit) || player.Stock(selectedOutfit) > 0 || isAlreadyOwned))
-		return false;
-
+	// Check special unique outfits, if you already have them.
 	int mapSize = selectedOutfit->Get("map");
 	if(mapSize > 0 && player.HasMapped(mapSize))
-		return false;
+		return "You have already mapped all the systems shown by this map, "
+			"so there is no reason to buy another.";
+
+	if(HasLicense(selectedOutfit->Name()))
+		return "You already have one of these licenses, "
+			"so there is no reason to buy another.";
 
-	// Determine what you will have to pay to buy this outfit.
-	int64_t cost = player.StockDepreciation().Value(selectedOutfit, day);
 	// Check that the player has any necessary licenses.
 	int64_t licenseCost = LicenseCost(selectedOutfit);
 	if(licenseCost < 0)
-		return false;
-	cost += licenseCost;
-	// If you have this in your cargo hold or in planetary storage, installing it is free.
-	if(cost > player.Accounts().Credits() && !isAlreadyOwned)
-		return false;
+		return "You cannot buy this outfit, because it requires a license that you don't have.";
 
-	if(HasLicense(selectedOutfit->Name()))
-		return false;
+	// Check if the outfit is available to get at all.
+	bool isInCargo = player.Cargo().Get(selectedOutfit);
+	bool isInStorage = player.Storage() && player.Storage()->Get(selectedOutfit);
+	bool isInStore = outfitter.Has(selectedOutfit) || player.Stock(selectedOutfit) > 0;
+	if(isInStorage
+		&& !(!onlyOwned && !isInStore && !playerShip))
+	{
+		// In storage, the outfit is certainly available to get
 
-	if(!playerShip)
+		// Except for this one case: 'b' does not move storage to cargo.
```

Comment:
Shouldn't be a blank line here, the first line of the comment should have a full stop.